ArduCopter/GCS_Mavlink.cpp

```diff
@@ -583,6 +583,49 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_packet(const mavlink_command_i
 #endif
         return MAV_RESULT_UNSUPPORTED;
 
+      case MAV_CMD_DO_REPOSITION:
+            {
+                const bool change_modes = ((int32_t)packet.param2 & MAV_DO_REPOSITION_FLAGS_CHANGE_MODE) == MAV_DO_REPOSITION_FLAGS_CHANGE_MODE;
+                if (!copter.flightmode->in_guided_mode() && !change_modes) {
+                    return MAV_RESULT_FAILED;
+                }
+
+                Location requested_position {};
+                requested_position.lat = packet.x;
+                requested_position.lng = packet.y;
+
+                // check the floating representation for overflow of altitude
+                if (fabsf(packet.z * 100.0f) >= 0x7fffff) {
+                    return MAV_RESULT_FAILED;
+                }
+                requested_position.alt = (int32_t)(packet.z * 100.0f);
+
+                // load option flags
```

Comment:
Can we use `mavlink_coordinate_frame_to_location_alt_frame` here?

Also, this looks a bit neater  IMO - assembling the packet at once:

https://github.com/ardupilot/ardupilot/blob/pr%2Fset-param-direct/libraries/GCS_MAVLink/GCS_Common.cpp#L4022

And speaking of that message; we're not doing the "0,0-means-here" thing.  I know this is "reposition" not "position", but it might be would could factor a lot of the extract-location-from-command-int code if we're consistent.

Reply:
Yeah, I had the same thought, but like the result denied I was trying to keep the code as tightly similiar to plane as possible. If we convert plane over I don't mind tweaking it, but I'd like these to look as similar as possible

Reply:
Let's fix Plane.